cluster/gce/util.sh

```diff
@@ -584,6 +584,26 @@ function kube-up {
   find-release-tars
   upload-server-tars
 
+  local running_in_terminal=false
+  if [ -t 1 ]; then
+    running_in_terminal=true
+  fi
+
+  if [[ ${running_in_terminal} == "true" || ${KUBE_UP_AUTOMATIC_CLEANUP} == "true" ]]; then 
```

Comment:
I don't think we should default to yes in any circumstance. The first time we accidentally recycle a cluster because someone buried `kube-up` in a script, that choice will haunt us. :/

I like the env variable.

Remember that as upgrade starts to be more of a reality, up/down should become relevant to testing and out-of-the-box / first-touch experience alone, rather than the constant experience, so convenience for things like teardown / recreate are less important as we go forward.